src/textual/reactive.py

```diff
@@ -50,6 +53,108 @@ class TooManyComputesError(ReactiveError):
 
 
+class WatchDecorator(Generic[WatchMethodType]):
+    """Watch decorator.
+
+    Decorate a method to make it a watcher.
+
+    """
+
+    def __init__(self, reactive: Reactive | None = None) -> None:
+        self._reactive = reactive
+
+    @overload
+    def __call__(self, *, init: bool = True) -> WatchDecorator[WatchMethodType]: ...
+
+    @overload
+    def __call__(self, method: WatchMethodType) -> WatchMethodType: ...
+
+    def __call__(
+        self, method: WatchMethodType | None = None, *, init: bool = True
+    ) -> WatchMethodType | WatchDecorator[WatchMethodType]:
+        _rich_traceback_omit = True
+        if method is None:
+            return self
+        assert self._reactive is not None
+        assert hasattr(method, "__name__")
+        if not method.__name__.startswith("watch_"):
+            if self._reactive._watch_method is not None:
+                raise RuntimeError(
+                    "Only a single method may be decorator with watch (per-reactive)."
+                )
+            self._reactive._watch_method = method
+            self._reactive._watch_method_init = init
+        return method
+
+
+class ComputeDecorator(Generic[ComputeMethodType]):
+    """Watch decorator.
+
+    Decorate a widget method to make it a compute method.
+
+    """
+
+    def __init__(self, reactive: Reactive | None = None) -> None:
+        self._reactive = reactive
+
+    @overload
+    def __call__(self, *, init: bool = True) -> ComputeDecorator[ComputeMethodType]: ...
+
+    @overload
+    def __call__(self, method: ComputeMethodType) -> ComputeMethodType: ...
+
+    def __call__(
+        self, method: ComputeMethodType | None = None, *, init: bool = True
+    ) -> ComputeMethodType | ComputeDecorator[ComputeMethodType]:
+        _rich_traceback_omit = True
+        if method is None:
+            return self
+        assert self._reactive is not None
+        assert hasattr(method, "__name__")
+        if not method.__name__.startswith("compute_"):
+            if self._reactive._compute_method is not None:
+                raise RuntimeError(
+                    "Only a single method may be decorated with compute (per-reactive)."
+                )
+            self._reactive._compute_method = method
+        return method
+
+
+class ValidateDecorator(Generic[ValidateMethodType]):
+    """Validate decorator.
+
+    Decorate a Widget method to make it a validator for the attribute/
+
+    """
+
+    def __init__(self, reactive: Reactive | None = None) -> None:
+        self._reactive = reactive
+
+    @overload
+    def __call__(
+        self, *, init: bool = True
+    ) -> ValidateDecorator[ValidateMethodType]: ...
+
+    @overload
+    def __call__(self, method: ValidateMethodType) -> ValidateMethodType: ...
+
+    def __call__(
+        self, method: ValidateMethodType | None = None, *, init: bool = True
+    ) -> ValidateMethodType | ValidateDecorator[ValidateMethodType]:
```

Comment:
It looks like `init` is not needed in the validate decorator, which means the overloads are superfluous.
If `init` _is_ needed, where is it used?